WordPress/Classes/Utility/ContextManager+ErrorHandling.swift

```diff
@@ -55,8 +55,38 @@ private extension NSExceptionName {
     static let coreDataSaveDerivedException = NSExceptionName("Unresolved Core Data save error (Derived Context)")
 }
 
-extension ContextManager {
-    @objc(handleSaveError:inContext:)
+extension LegacyContextFactory {
+
+    /// A wrapper of `internalSave(_:)` to expose it as an Objective-C API for `LegacyContextFactory` to call.
+    @objc func internalSaveContext(_ context: NSManagedObjectContext) {
+        internalSave(context)
+    }
+
+}
+
+extension ManagedObjectContextFactory {
+
+    func internalSave(_ context: NSManagedObjectContext) {
+        let inserted = Array(context.insertedObjects)
+        do {
+            try context.obtainPermanentIDs(for: inserted)
+        } catch {
+            DDLogError("Error obtaining permanent object IDs for \(inserted), \(error)")
+        }
+
+        if context.hasChanges {
+            do {
+                try context.save()
+            } catch {
+                handleSaveError(error as NSError, in: context)
+            }
+        }
```

Comment:
While this is a direct translation of the original, IMHO we should sneak in a slight performance improvement while we're at it?

```suggestion
        guard context.hasChanges else {
            return
        }
        
        let inserted = Array(context.insertedObjects)
        do {
            try context.obtainPermanentIDs(for: inserted)
        } catch {
            DDLogError("Error obtaining permanent object IDs for \(inserted), \(error)")
        }

        do {
            try context.save()
        } catch {
            handleSaveError(error as NSError, in: context)
        }

```

Reply:
Good call. I've made the change in https://github.com/wordpress-mobile/WordPress-iOS/pull/19219/commits/be515de98db68aee6d88bdbccaf983bd766d296d